subxt/src/blocks/extrinsic_types.rs

```diff
@@ -660,24 +660,22 @@ impl<'a, T: Config> ExtrinsicSignedExtensions<'a, T> {
         })
     }
 
-    fn find_by_name(&self, name: &str) -> Option<ExtrinsicSignedExtension<'_, T>> {
-        let signed_extension = self
-            .iter()
-            .find_map(|e| e.ok().filter(|e| e.name() == name))?;
-        Some(signed_extension)
-    }
-
     /// Searches through all signed extensions to find a specific one.
     /// If the Signed Extension is not found `Ok(None)` is returned.
     /// If the Signed Extension is found but decoding failed `Err(_)` is returned.
     pub fn find<S: SignedExtension<T>>(&self) -> Result<Option<S::Decoded>, Error> {
-        self.find_by_name(S::NAME)
-            .map(|s| {
-                s.as_signed_extra::<S>().map(|e| {
-                    e.expect("signed extra name is correct, because it was found before; qed.")
-                })
-            })
-            .transpose()
+        for ext in self.iter() {
+            let Ok(ext) = ext else { continue };
```

Comment:
The error is ignored here, can it be returned as an error if no other valid signed extension was found?

Or what's reason about that?

Reply:
That's a good point; initially I just skipped any iteration error with the reaosning that from `find`s perspective, we only care about an error if we find a signed ext and then can't decode it into the type we were hoping, and any iteration errors were sortof unrelated.

But that was probably a bit silly, and anyway, if we do hit an iteration error, we'll bail from `iter` anyway, so we may as well just return it!